src/main/java/com/epam/javacourses2016/task1/SolverTask1.java

```diff
@@ -16,7 +20,43 @@
      * @return Список строк, прочитанных из входного файла в прямом порядке.
      */
     public List<String> reverseFile(File input, File output) {
-        //TODO: realize solver of a task
-        return null;
+        List<String> strings = readLines(input);
+        Collections.reverse(strings);
+        writeLines(output, strings);
+        Collections.reverse(strings);
+        return strings;
+    }
+
+    private List<String> readLines(File input) {
+        ArrayList<String> strings = new ArrayList<>();
+        String s = null;
+        char[] buf = new char[1024];
+        try (FileReader fr = new FileReader(input); BufferedReader br = new BufferedReader(fr)) {
+
+            while((s = br.readLine()) != null) {
+                strings.add(s);
+            }
+        }
+        catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        return strings;
+    }
+
+    private void writeLines(File output, List<String> lines) {
+        String empty = "";
+        if (output != null && lines != null && lines.size() > 0)
```

Comment:
If you want to introduce predicate, throw on negative condition
Now you method do nothing with bed input, Java way is to fail when something is wrong